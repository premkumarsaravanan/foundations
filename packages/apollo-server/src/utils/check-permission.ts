import logger from '../logger'
import { ServerContext } from '../app'

export const checkPermission = (context: ServerContext) => {
  logger.info('checkPermission', { context })
  return true
}
