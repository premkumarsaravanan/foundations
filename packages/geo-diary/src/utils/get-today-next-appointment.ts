import dayjs from 'dayjs'
import { ExtendedAppointmentModel } from '@/types/core'

export function getTodayNextAppointment(appointments: ExtendedAppointmentModel[]) {
  return appointments
    .filter(
      appointment =>
        !appointment.cancelled &&
        dayjs(appointment.start).isSame(dayjs(), 'day') &&
        dayjs().isBefore(dayjs(appointment.start)),
    )
    .sort((a, b) => {
      return (dayjs(a.start).isAfter(dayjs(b.start)) ? 1 : -1)[0]
    })
}
