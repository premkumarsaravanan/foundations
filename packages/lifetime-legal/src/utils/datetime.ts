import dayjs from 'dayjs'
import { toUTCTime, toLocalTime, DATE_TIME_FORMAT } from '@reapit/elements'
// @ts-ignore #49 Breaking changes to API
import { ContactIdentityCheckModel } from '@reapit/foundations-ts-definitions'

/**
 * This function created by backend required parse time to UTC before submit
 * @param identityChecks is ContactIdentityCheckModel
 */
export const changeTimeZoneLocalForIdentityCheck = (
  identityChecks: ContactIdentityCheckModel,
): ContactIdentityCheckModel => {
  const documents = identityChecks.documents?.map(document => {
    return {
      ...document,
      expiry: toLocalTime(document.expiry as dayjs.ConfigType, DATE_TIME_FORMAT.RFC3339),
    }
  })
  const newIdentityChecks = {
    ...identityChecks,
    documents,
    checkDate: toLocalTime(identityChecks.checkDate as dayjs.ConfigType, DATE_TIME_FORMAT.RFC3339),
  }
  return newIdentityChecks
}

/**
 * This function created by backend required parse time to UTC before submit
 * @param identityChecks is ContactIdentityCheckModel
 */
export const changeTimeZoneUTCForIdentityCheck = (
  identityChecks: ContactIdentityCheckModel,
): ContactIdentityCheckModel => {
  const documents = identityChecks.documents?.map(document => {
    return {
      ...document,
      expiry: toUTCTime(document.expiry as dayjs.ConfigType),
    }
  })
  const newIdentityChecks = {
    ...identityChecks,
    documents,
    checkDate: toUTCTime(identityChecks.checkDate as dayjs.ConfigType),
  }
  return newIdentityChecks
}
