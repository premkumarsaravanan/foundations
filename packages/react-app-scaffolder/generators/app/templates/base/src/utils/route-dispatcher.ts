import { RouteValue, StringMap } from '../types/core'
import Routes from '../constants/routes'
import store from '../core/store'
import { homeRequestData } from '../actions/home'

const routeDispatcher = async (route: RouteValue, params?: StringMap) => {
  switch (route) {
    case Routes.HOME:
      store.dispatch(homeRequestData())
      break
    default:
      console.error('Route not found, nothing to fetch')
  }
}

export default routeDispatcher
