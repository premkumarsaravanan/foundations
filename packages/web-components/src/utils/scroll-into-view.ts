function scrollIntoView(
  element: Element,
  style: string,
  unit: string,
  from: number,
  to: number,
  time: number,
  prop: boolean
) {
  if (!element) {
    return
  }
  const start = new Date().getTime()
  const timer = setInterval(function() {
    const step = Math.min(1, (new Date().getTime() - start) / time)
    if (prop) {
      // @ts-ignore
      element[style] = from + step * (to - from) + unit
    } else {
      // @ts-ignore
      element.style[style] = from + step * (to - from) + unit
    }
    if (step === 1) {
      clearInterval(timer)
    }
  }, 25)
  if (prop) {
    //
    element[style] = from + unit
  } else {
    // @ts-ignore
    element.style[style] = from + unit
  }
}

export default scrollIntoView
