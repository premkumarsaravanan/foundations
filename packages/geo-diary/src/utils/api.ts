import { getAccessToken } from '@/utils/session'
import { APPOINTMENTS_HEADERS } from '@/constants/api'

export const initAuthorizedRequestHeaders = async () => ({
  ...APPOINTMENTS_HEADERS,
  Authorization: `Bearer ${await getAccessToken()}`,
})
