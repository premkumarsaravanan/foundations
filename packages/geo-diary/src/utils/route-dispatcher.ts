import { appointmentsRequestData } from '@/actions/appointments'
import { RouteValue, StringMap } from '@/types/core'
import Routes from '@/constants/routes'
import store from '@/core/store'
import { getAccessToken } from '@/utils/session'

const routeDispatcher = async (route: RouteValue, _params?: StringMap) => {
  await getAccessToken()

  switch (route) {
    case Routes.HOME:
      store.dispatch(appointmentsRequestData({ time: 'Today' }))
      break
    default:
      console.error('Route not found, nothing to fetch')
  }
}

export default routeDispatcher
