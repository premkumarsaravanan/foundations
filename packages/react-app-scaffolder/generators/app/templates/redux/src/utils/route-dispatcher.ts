import { RouteValue } from '@/types/core'
import { getAccessToken } from '@/utils/session'

const routeDispatcher = async (route: RouteValue) => {
  await getAccessToken()

  switch (route) {
    default:
      console.error('Route not found, nothing to fetch')
  }
}

export default routeDispatcher
