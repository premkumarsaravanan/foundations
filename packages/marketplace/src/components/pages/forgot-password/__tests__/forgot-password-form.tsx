import React from 'react'
import { shallow } from 'enzyme'
import { getMockRouterProps } from '@/utils/mock-helper'
import { mockFormikAction, mockWithFormik } from '@/utils/mock-formik'
import {
  ForgotPasswordForm,
  mapPropsToValues,
  ForgotPasswordValues,
  handleSubmitForgotPassword,
  ForgotPasswordFormProps,
  SuccessForgetPasswordContent
} from '../forgot-password-form'

describe('ForgotPasswordForm', () => {
  it('should match snapshot', () => {
    const mockProps = {
      ...mockWithFormik({ email: '1' }),
      ...getMockRouterProps({}),
      getFieldHelpers: jest.fn()
    } as ForgotPasswordFormProps
    const wrapper = shallow(<ForgotPasswordForm {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('[dataTest="email"]')).toHaveLength(1)
  })

  it('should match snapshot', () => {
    const mockProps = {
      ...mockWithFormik({ email: '1' }),
      ...getMockRouterProps({}),
      location: {
        hash: '',
        key: '',
        pathname: '',
        search: '?isError=1',
        state: {}
      },
      getFieldHelpers: jest.fn()
    } as ForgotPasswordFormProps
    const wrapper = shallow(<ForgotPasswordForm {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('[dataTest="email"]')).toHaveLength(1)
  })

  it('should match snapshot', () => {
    const mockProps = {
      ...mockWithFormik({ email: '1' }),
      ...getMockRouterProps({}),
      location: {
        hash: '',
        key: '',
        pathname: '',
        search: '?isSuccess=1',
        state: {}
      },
      getFieldHelpers: jest.fn()
    } as ForgotPasswordFormProps
    const wrapper = shallow(<ForgotPasswordForm {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapPropsToValues', () => {
    it('should run correctly', () => {
      const result = mapPropsToValues()
      expect(result).toEqual({ email: '' })
    })
  })

  describe('mapPropsToValues', () => {
    it('should run correctly', () => {
      const result = mapPropsToValues()
      expect(result).toEqual({ email: '' })
    })
  })

  describe('handleSubmitForgotPassword', () => {
    it('should call setSubmitting with false', done => {
      const mockValues = {
        email: 'abc@gmail.com'
      } as ForgotPasswordValues
      const mockProps = {
        ...getMockRouterProps({}),
        submitEmail: jest.fn()
      }
      const mockFormik = {
        ...mockFormikAction
      }
      setTimeout(() => {
        handleSubmitForgotPassword(mockValues, { ...mockFormik, props: mockProps })
        expect(mockProps.submitEmail).toBeCalled()
        done()
      }, 1000)
    })
  })
})

describe('SuccessForgetPasswordContent', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SuccessForgetPasswordContent />)
    expect(wrapper).toMatchSnapshot()
  })
})
