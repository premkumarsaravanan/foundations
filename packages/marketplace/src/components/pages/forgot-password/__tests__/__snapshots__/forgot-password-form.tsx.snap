// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordForm should match snapshot 1`] = `
<Form>
  <FormSubHeading>
    Please enter your email address to reset your password
  </FormSubHeading>
  <Grid>
    <GridItem>
      <Input
        dataTest="email"
        id="email"
        labelText="Email"
        name="email"
        type="email"
      />
    </GridItem>
  </Grid>
  <FlexContainerResponsive>
    <Button
      disabled={false}
      loading={true}
      type="submit"
      variant="primary"
    >
      Submit
    </Button>
  </FlexContainerResponsive>
</Form>
`;

exports[`ForgotPasswordForm should match snapshot 2`] = `
<Form>
  <FormSubHeading>
    Please enter your email address to reset your password
  </FormSubHeading>
  <Grid>
    <GridItem>
      <Input
        dataTest="email"
        id="email"
        labelText="Email"
        name="email"
        type="email"
      />
    </GridItem>
  </Grid>
  <Alert
    message="Your forgotten password request has failed. Please try again or contact our support desk"
    type="danger"
  />
  <FlexContainerResponsive>
    <Button
      disabled={false}
      loading={true}
      type="submit"
      variant="primary"
    >
      Submit
    </Button>
  </FlexContainerResponsive>
</Form>
`;

exports[`ForgotPasswordForm should match snapshot 3`] = `<Component />`;

exports[`SuccessForgetPasswordContent should match snapshot 1`] = `
<FlexContainerBasic
  centerContent={true}
  flexColumn={true}
  hasPadding={true}
>
  <H5
    isCentered={true}
  >
    Success
  </H5>
  <span>
    Please check your inbox for an email we have just sent you with instructions for how to reset your password.
  </span>
  <br />
  <Link
    to="/developer/login"
  >
    Click here to return to the login page
  </Link>
</FlexContainerBasic>
`;
